### PR TITLE
Initial commit of relenv linux packages

### PR DESCRIPTION
**Pull request by dwoz**
### What does this PR do?

* Pulling in debian packaging from salt-pkg back into the main salt repository.
* Using relenv for 'onedir' packaging approach

### What issues does this PR fix or reference?
Fixes:

### Previous Behavior
Remove this section if not relevant

### New Behavior
Remove this section if not relevant

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes/No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

**Comment by dwoz** [comment]
re-run all

File: pkg/deb/debian/changelog [hunk 1]
```diff
@@ -0,0 +1,5 @@
+salt (3006.0) stable; urgency=medium
+
+  * Build of Salt {{version}} with Tiamat
```
Comment (Ch3LL) [inline pkg/deb/debian/changelog:3]:
```suggestion
  * Build of Salt {{version}} with relenv
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: pkg/salt-api.service [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=The Salt API
-Documentation=man:salt-api(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
+Documentation=man:salt-api(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
```
Comment (s0undt3ch) [inline pkg/salt-api.service:3]:
```suggestion
Documentation=man:salt-api(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
```
This line had the correct URL

File: pkg/salt-master.service [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=The Salt Master Server
-Documentation=man:salt-master(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
+Documentation=man:salt-master(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
```
Comment (s0undt3ch) [inline pkg/salt-master.service:3]: This too

File: pkg/salt-minion.service [hunk 4]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=The Salt Minion
-Documentation=man:salt-minion(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
+Documentation=man:salt-minion(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
```
Comment (s0undt3ch) [inline pkg/salt-minion.service:3]: And this

File: pkg/salt-proxy@.service [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [Unit]
 Description=salt-proxy service for %i
-Documentation=man:salt-proxy(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
+Documentation=man:salt-proxy(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
```
Comment (s0undt3ch) [inline pkg/salt-proxy@.service:3]: And this

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: pkg/README [hunk 6]
```diff
@@ -0,0 +1,11 @@
+Salt packaging resources.
+
+
+The contents of this directory are as follows
+
+common/ - Files common to multiple pacakges.
```
Comment (s0undt3ch) [inline pkg/README:6]:
```suggestion
common/ - Files common to multiple packages.
```

File: pkg/debian/changelog [hunk 7]
```diff
@@ -0,0 +1,5 @@
+salt (3006.0) stable; urgency=medium
+
+  * Build of Salt {{version}} with Tiamat
```
Comment (s0undt3ch) [inline pkg/debian/changelog:3]:
```suggestion
  * Build of Salt {{version}} with Relenv
```

File: pkg/debian/control [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+Priority: optional
+Maintainer: Debian Salt Team <pkg-salt-team@lists.alioth.debian.org>
+Uploaders: Joe Healy <joehealy@gmail.com>,
+           Franklin G Mendoza <franklin.g.mendoza@gmail.com>,
+           Andriy Senkovych <jolly_roger@itblog.org.ua>,
+           David Murphy <dmurphy@saltstack.com>
+Build-Depends: bash-completion,
+               debhelper (>= 10)
+Standards-Version: 4.1.3
+Homepage: http://saltstack.org/
```
Comment (s0undt3ch) [inline pkg/debian/control:12]:
```suggestion
Homepage: http://saltproject.io
```
?

File: pkg/old/README.md [hunk 9]
```diff
@@ -0,0 +1,4 @@
+The things found in this directory have been depericated and will be removed in
+the 3008.0 release of salt. The Salt maintainers are not aware of anyone
+relying these files. Please file a bug report if you are relying any files that
```
Comment (s0undt3ch) [inline pkg/old/README.md:3]:
```suggestion
relying on these files. Please file a bug report if you are relying any files that
```

**Review by s0undt3ch — CHANGES_REQUESTED** [review]

File: pkg/old/README.md [hunk 10]
```diff
@@ -0,0 +1,4 @@
+The things found in this directory have been depericated and will be removed in
```
Comment (Ch3LL) [inline pkg/old/README.md:1]:
```suggestion
The things found in this directory have been deprecated and will be removed in
```

**Review by Ch3LL — CHANGES_REQUESTED** [review]

**Review by s0undt3ch — DISMISSED** [review]

**Review by Ch3LL — APPROVED** [review]

**Pull request merged**